lightning-persister/src/lib.rs

```diff
@@ -108,39 +102,64 @@ impl FilesystemPersister {
 		util::write_to_file(path, "manager".to_string(), manager)
 	}
 
-	#[cfg(test)]
-	fn load_channel_data<Keys: KeysInterface>(&self, keys: &Keys) ->
-		Result<HashMap<OutPoint, ChannelMonitor<Keys::Signer>>, ChannelMonitorUpdateErr> {
-			if let Err(_) = fs::create_dir_all(self.path_to_monitor_data()) {
-				return Err(ChannelMonitorUpdateErr::PermanentFailure);
+	/// Read `ChannelMonitor`s from disk.
+	pub fn read_channelmonitors<Signer: Sign, K: Deref> (
+		&self, keys_manager: K
+	) -> Result<HashMap<OutPoint, (BlockHash, ChannelMonitor<Signer>)>, std::io::Error>
+		where K::Target: KeysInterface<Signer=Signer> + Sized
+	{
+		let path = self.path_to_monitor_data();
+		if !Path::new(&path).exists() {
+			return Ok(HashMap::new());
+		}
+		let mut outpoint_to_channelmonitor = HashMap::new();
+		for file_option in fs::read_dir(path).unwrap() {
+			let file = file_option.unwrap();
+			let owned_file_name = file.file_name();
+			let filename = owned_file_name.to_str();
+			if !filename.is_some() || !filename.unwrap().is_ascii() || filename.unwrap().len() < 65 {
+				return Err(std::io::Error::new(
+					std::io::ErrorKind::Other,
+					"Invalid ChannelMonitor file name",
+				));
 			}
-			let mut res = HashMap::new();
-			for file_option in fs::read_dir(self.path_to_monitor_data()).unwrap() {
-				let file = file_option.unwrap();
-				let owned_file_name = file.file_name();
-				let filename = owned_file_name.to_str();
-				if !filename.is_some() || !filename.unwrap().is_ascii() || filename.unwrap().len() < 65 {
-					return Err(ChannelMonitorUpdateErr::PermanentFailure);
-				}
-
-				let txid = Txid::from_hex(filename.unwrap().split_at(64).0);
-				if txid.is_err() { return Err(ChannelMonitorUpdateErr::PermanentFailure); }
 
-				let index = filename.unwrap().split_at(65).1.split('.').next().unwrap().parse();
-				if index.is_err() { return Err(ChannelMonitorUpdateErr::PermanentFailure); }
-
-				let contents = fs::read(&file.path());
-				if contents.is_err() { return Err(ChannelMonitorUpdateErr::PermanentFailure); }
+			let txid = Txid::from_hex(filename.unwrap().split_at(64).0);
+			if txid.is_err() {
+				return Err(std::io::Error::new(
+					std::io::ErrorKind::Other,
+					"Invalid tx ID in filename",
+				));
+			}
 
-				if let Ok((_, loaded_monitor)) =
-					<(BlockHash, ChannelMonitor<Keys::Signer>)>::read(&mut Cursor::new(&contents.unwrap()), keys) {
-						res.insert(OutPoint { txid: txid.unwrap(), index: index.unwrap() }, loaded_monitor);
-					} else {
-						return Err(ChannelMonitorUpdateErr::PermanentFailure);
-					}
+			let index = filename.unwrap().split_at(65).1.split('.').next().unwrap().parse();
```

Comment:
nit: Wait, we don't actually generate files with `.*` at the end, do we? I guess we could, but don't need the split, I guess?